src/ILCompiler.Compiler/src/Compiler/NameMangler.cs

```diff
@@ -345,5 +348,40 @@ private string ComputeMangledFieldName(FieldDesc field)
 
             return mangledName;
         }
+
+        private ImmutableDictionary<string, string> _mangledStringLiterals = ImmutableDictionary<string, string>.Empty;
+
+        public string GetMangledStringName(string literal)
+        {
+            string mangledName;
+            if (_mangledStringLiterals.TryGetValue(literal, out mangledName))
+                return mangledName;
+
+            return ComputeMangledStringLiteralName(literal);
+        }
+
+        private string ComputeMangledStringLiteralName(string literal)
+        {
+            string mangledName = SanitizeName(literal);
+
+            if (mangledName.Length > 30)
+                mangledName = mangledName.Substring(0, 30);
+
+            if (mangledName != literal)
+            {
+                if (_sha1 == null)
```

Comment:
I don't fully understand the SHA1 dance. I understand it's an attempt to avoid collisions, but it doesn't avoid them, just makes them less likely. At that point we might want to use just the SHA1 itself.

Can you make sure use of SHA1 (even for non-cryptographic purposes) doesn't need crypto signoffs?